module,execution_time (sec),success,timestamp,error,error_short
cowidev.vax.batch.argentina,11.28,True,2023-05-27T07:00:17,,
cowidev.vax.batch.australia,,,2023-05-27T07:00:06,,
cowidev.vax.batch.austria_old,,,2023-05-27T07:00:06,,
cowidev.vax.batch.belgium,10.1,True,2023-05-27T07:00:16,,
cowidev.vax.batch.bolivia,4.85,True,2023-05-27T07:00:11,,
cowidev.vax.batch.canada,9.56,False,2023-05-27T07:02:05,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 282, in main
    Canada().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 270, in export
    df = df.pipe(self.pipeline, df_man)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 250, in pipeline
    .pipe(self.pipe_make_monotonic)[
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 230, in pipe_make_monotonic
    df = df.pipe(self.make_monotonic, max_removed_rows=self.max_removed_rows)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/utils/base.py"", line 120, in make_monotonic
    return mkm(
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/utils.py"", line 23, in make_monotonic
    return make_monotonic_new(df, column_date, column_metrics, max_removed_rows)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/utils.py"", line 96, in make_monotonic_new
    raise Exception(
Exception: 1 chunks of lengths 36 would be NaNed for metric people_vaccinated. That is more than maximum allowed (22) by make_monotonic() - check the data. Check dates ['2023-01-26']. Location(s) affected: {'Canada'}
",1 chunks of lengths 36 would be NaNed for metric people_vaccinated. That is more than maximum allowed (22) by make_monotonic() - check the data. Check dates ['2023-01-26']. Location(s) affected: {'Canada'}
cowidev.vax.batch.chile,0.95,True,2023-05-27T07:00:07,,
cowidev.vax.batch.czechia,26.09,True,2023-05-27T07:00:32,,
cowidev.vax.batch.denmark,,,2023-05-27T07:00:06,,
cowidev.vax.batch.ecdc,93.23,True,2023-05-27T07:01:39,,
cowidev.vax.batch.ecuador,,,2023-05-27T07:00:06,,
cowidev.vax.batch.estonia,0.68,True,2023-05-27T07:00:07,,
cowidev.vax.batch.france,7.9,True,2023-05-27T07:00:14,,
cowidev.vax.batch.germany,0.34,True,2023-05-27T07:00:06,,
cowidev.vax.batch.greece,19.93,True,2023-05-27T07:00:26,,
cowidev.vax.batch.hong_kong,13.56,True,2023-05-27T07:00:19,,
cowidev.vax.batch.indonesia,,,2023-05-27T07:00:06,,
cowidev.vax.batch.ireland,7.24,True,2023-05-27T07:00:13,,
cowidev.vax.batch.israel,9.28,True,2023-05-27T07:00:15,,
cowidev.vax.batch.italy,8.89,True,2023-05-27T07:00:15,,
cowidev.vax.batch.japan,9.61,False,2023-05-27T07:02:14,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 375, in main
    Japan().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 359, in export
    df = self.read().pipe(self.pipeline_base)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 71, in read
    df_latest = self.read_latest().pipe(self.pipe_read_latest)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 100, in read_latest
    self._read_latest_booster(),
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 211, in _read_latest_booster
    df = self._read_xlsx(
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 233, in _read_xlsx
    raise ValueError(f""Unknown sheets: {sheets_unknown}"")
ValueError: Unknown sheets: {'5.8以降接種_一般接種（３回目）', '5.8以降接種_一般接種（４回目）', '5.8以降接種_総接種回数', '５回目_総接種回数', '5.8以降接種_一般接種（６回目）', '５回目_一般接種回数', '６回目_総接種回数', '６回目_一般接種回数', '5.8以降接種_一般接種（５回目）'}
","Unknown sheets: {'5.8以降接種_一般接種（３回目）', '5.8以降接種_一般接種（４回目）', '5.8以降接種_総接種回数', '５回目_総接種回数', '5.8以降接種_一般接種（６回目）', '５回目_一般接種回数', '６回目_総接種回数', '６回目_一般接種回数', '5.8以降接種_一般接種（５回目）'}"
cowidev.vax.batch.jersey,5.76,True,2023-05-27T07:00:12,,
cowidev.vax.batch.latvia,65.26,True,2023-05-27T07:01:11,,
cowidev.vax.batch.lithuania,7.28,True,2023-05-27T07:00:13,,
cowidev.vax.batch.luxembourg,3.35,True,2023-05-27T07:00:09,,
cowidev.vax.batch.malaysia,1.13,True,2023-05-27T07:00:07,,
cowidev.vax.batch.malta,1.38,True,2023-05-27T07:00:07,,
cowidev.vax.batch.netherlands,,,2023-05-27T07:00:06,,
cowidev.vax.batch.new_zealand,13.77,True,2023-05-27T07:00:20,,
cowidev.vax.batch.norway,0.62,True,2023-05-27T07:00:07,,
cowidev.vax.batch.peru,5.76,True,2023-05-27T07:00:12,,
cowidev.vax.batch.portugal,,,2023-05-27T07:00:06,,
cowidev.vax.batch.romania,33.78,True,2023-05-27T07:00:40,,
cowidev.vax.batch.saudi_arabia,9.43,True,2023-05-27T07:00:15,,
cowidev.vax.batch.singapore,5.34,True,2023-05-27T07:00:11,,
cowidev.vax.batch.slovakia,51.37,True,2023-05-27T07:00:57,,
cowidev.vax.batch.slovenia,3.36,True,2023-05-27T07:00:09,,
cowidev.vax.batch.south_korea,69.48,True,2023-05-27T07:01:15,,
cowidev.vax.batch.spc,33.36,True,2023-05-27T07:00:40,,
cowidev.vax.batch.sweden,23.61,True,2023-05-27T07:00:30,,
cowidev.vax.batch.switzerland,84.49,True,2023-05-27T07:01:31,,
cowidev.vax.batch.trinidad_and_tobago,4.89,True,2023-05-27T07:00:12,,
cowidev.vax.batch.ukraine,,,2023-05-27T07:00:07,,
cowidev.vax.batch.united_kingdom,11.05,True,2023-05-27T07:00:18,,
cowidev.vax.batch.united_states,16.89,True,2023-05-27T07:00:24,,
cowidev.vax.batch.uruguay,6.54,True,2023-05-27T07:00:16,,
cowidev.vax.batch.zimbabwe,2.9,True,2023-05-27T07:00:12,,
cowidev.vax.incremental.africacdc,,,2023-05-27T07:00:11,,
cowidev.vax.incremental.antigua_barbuda,,,2023-05-27T07:00:11,,
cowidev.vax.incremental.aruba,2.44,True,2023-05-27T07:00:13,,
cowidev.vax.incremental.austria,,,2023-05-27T07:00:11,,
cowidev.vax.incremental.azerbaijan,,,2023-05-27T07:00:11,,
cowidev.vax.incremental.bahrain,,,2023-05-27T07:00:11,,
cowidev.vax.incremental.bangladesh,46.45,True,2023-05-27T07:00:58,,
cowidev.vax.incremental.barbados,,,2023-05-27T07:00:12,,
cowidev.vax.incremental.brazil,1.77,True,2023-05-27T07:00:13,,
cowidev.vax.incremental.bulgaria,5.72,True,2023-05-27T07:00:17,,
cowidev.vax.incremental.china,21.11,True,2023-05-27T07:00:33,,
cowidev.vax.incremental.costa_rica,,,2023-05-27T07:00:12,,
cowidev.vax.incremental.croatia,1.5,True,2023-05-27T07:00:14,,
cowidev.vax.incremental.cuba,15.47,True,2023-05-27T07:00:29,,
cowidev.vax.incremental.curacao,,,2023-05-27T07:00:13,,
cowidev.vax.incremental.cyprus,1.81,True,2023-05-27T07:00:15,,
cowidev.vax.incremental.dominican_republic,13.01,False,2023-05-27T07:02:27,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 112, in main
    DominicanRepublic().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 68, in export
    data = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 26, in read
    metrics = self._parse_metrics(driver)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 35, in _parse_metrics
    metric_candidates = _find_potential_metrics(driver)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 91, in _find_potential_metrics
    h3 = _find_h3(driver)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 81, in _find_h3
    raise ValueError(""More than one element found!"")
ValueError: More than one element found!
",More than one element found!
cowidev.vax.incremental.el_salvador,5.53,True,2023-05-27T07:00:19,,
cowidev.vax.incremental.faeroe_islands,1.38,True,2023-05-27T07:00:15,,
cowidev.vax.incremental.fiji,0.0,True,2023-05-27T07:00:14,,
cowidev.vax.incremental.finland,0.65,True,2023-05-27T07:00:14,,
cowidev.vax.incremental.gabon,,,2023-05-27T07:00:14,,
cowidev.vax.incremental.georgia,2.73,True,2023-05-27T07:00:17,,
cowidev.vax.incremental.greenland,,,2023-05-27T07:00:15,,
cowidev.vax.incremental.guatemala,,,2023-05-27T07:00:15,,
cowidev.vax.incremental.guernsey,,,2023-05-27T07:00:15,,
cowidev.vax.incremental.hungary,81.58,True,2023-05-27T07:01:36,,
cowidev.vax.incremental.iceland,1.39,True,2023-05-27T07:00:16,,
cowidev.vax.incremental.india,1.42,True,2023-05-27T07:00:16,,
cowidev.vax.incremental.iran,,,2023-05-27T07:00:15,,
cowidev.vax.incremental.isle_of_man,2.78,True,2023-05-27T07:00:18,,
cowidev.vax.incremental.jamaica,6.39,True,2023-05-27T07:00:22,,
cowidev.vax.incremental.kazakhstan,9.01,True,2023-05-27T07:00:25,,
cowidev.vax.incremental.kosovo,,,2023-05-27T07:00:16,,
cowidev.vax.incremental.kyrgyzstan,2.04,True,2023-05-27T07:00:18,,
cowidev.vax.incremental.laos,5.78,True,2023-05-27T07:00:22,,
cowidev.vax.incremental.macao,7.01,False,2023-05-27T07:02:34,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/macao.py"", line 90, in main
    Macao().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/macao.py"", line 75, in export
    data = self.read()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/macao.py"", line 23, in read
    iframe_url = self._get_iframe_url(driver)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/macao.py"", line 31, in _get_iframe_url
    elem = driver.find_element_by_id(""ICovid19Monitor"")
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py"", line 360, in find_element_by_id
    return self.find_element(by=By.ID, value=id_)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py"", line 976, in find_element
    return self.execute(Command.FIND_ELEMENT, {
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py"", line 321, in execute
    self.error_handler.check_response(response)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/selenium/webdriver/remote/errorhandler.py"", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {""method"":""css selector"",""selector"":""[id=""ICovid19Monitor""]""}
  (Session info: headless chrome=113.0.5672.126)

","Message: no such element: Unable to locate element: {""method"":""css selector"",""selector"":""[id=""ICovid19Monitor""]""}
  (Session info: headless chrome=113.0.5672.126)
"
cowidev.vax.incremental.moldova,,,2023-05-27T07:00:17,,
cowidev.vax.incremental.monaco,89.22,True,2023-05-27T07:01:46,,
cowidev.vax.incremental.mongolia,0.74,True,2023-05-27T07:00:18,,
cowidev.vax.incremental.montenegro,0.5,True,2023-05-27T07:00:18,,
cowidev.vax.incremental.morocco,,,2023-05-27T07:00:18,,
cowidev.vax.incremental.myanmar,,,2023-05-27T07:00:18,,
cowidev.vax.incremental.nepal,,,2023-05-27T07:00:18,,
cowidev.vax.incremental.north_macedonia,,,2023-05-27T07:00:18,,
cowidev.vax.incremental.northern_cyprus,4.95,True,2023-05-27T07:00:23,,
cowidev.vax.incremental.paho,82.65,False,2023-05-27T07:03:57,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 195, in main
    PAHO().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 190, in export
    df = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 167, in pipeline
    df.pipe(self.pipe_check_columns)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 122, in pipe_check_columns
    raise ValueError(f""Missing column fields: {columns_missing}. Present columns are: {df.columns}"")
ValueError: Missing column fields: {'First dose', 'Single dose', '1st additional dose', 'Second dose', '2nd additional dose', 'Country/ Territory', 'Country code', 'Total doses'}. Present columns are: Index(['date,location,total_vaccinations,people_vaccinated,people_fully_vaccinated,total_boosters,new_vaccinations,new_vaccinations_interpolated,new_people_vaccinated_interpolated,new_vaccinations_smoothed,new_people_vaccinated_smoothed', 'date'], dtype='object')
","Missing column fields: {'First dose', 'Single dose', '1st additional dose', 'Second dose', '2nd additional dose', 'Country/ Territory', 'Country code', 'Total doses'}. Present columns are: Index(['date,location,total_vaccinations,people_vaccinated,people_fully_vaccinated,total_boosters,new_vaccinations,new_vaccinations_interpolated,new_people_vaccinated_interpolated,new_vaccinations_smoothed,new_people_vaccinated_smoothed', 'date'], dtype='object')"
cowidev.vax.incremental.pakistan,,,2023-05-27T07:00:18,,
cowidev.vax.incremental.philippines,,,2023-05-27T07:00:18,,
cowidev.vax.incremental.poland,1.74,True,2023-05-27T07:00:20,,
cowidev.vax.incremental.qatar,,,2023-05-27T07:00:18,,
cowidev.vax.incremental.russia,21.78,True,2023-05-27T07:00:40,,
cowidev.vax.incremental.saint_lucia,,,2023-05-27T07:00:18,,
cowidev.vax.incremental.serbia,9.66,True,2023-05-27T07:00:28,,
cowidev.vax.incremental.south_africa,,,2023-05-27T07:00:19,,
cowidev.vax.incremental.spain,19.67,True,2023-05-27T07:00:39,,
cowidev.vax.incremental.sri_lanka,,,2023-05-27T07:00:19,,
cowidev.vax.incremental.suriname,14.98,True,2023-05-27T07:00:34,,
cowidev.vax.incremental.taiwan,8.6,False,2023-05-27T07:04:06,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 184, in main
    Taiwan().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 170, in export
    data = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 37, in read
    df = self._parse_table(url_pdf)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 76, in _parse_table
    raise ValueError(f""There are some unknown columns: {cols}"")
ValueError: There are some unknown columns: Index(['廠牌', '劑次', '4 月接種人次', '累計至 4 月接種人次'], dtype='object')
","There are some unknown columns: Index(['廠牌', '劑次', '4 月接種人次', '累計至 4 月接種人次'], dtype='object')"
cowidev.vax.incremental.thailand,6.89,True,2023-05-27T07:00:27,,
cowidev.vax.incremental.turkey,,,2023-05-27T07:00:22,,
cowidev.vax.incremental.united_arab_emirates,,,2023-05-27T07:00:22,,
cowidev.vax.incremental.vietnam,,,2023-05-27T07:00:22,,
cowidev.vax.incremental.who,26.36,True,2023-05-27T07:00:48,,
cowidev.vax.incremental.zambia,1.93,True,2023-05-27T07:00:24,,
